src/sql/parts/modelComponents/componentBase.ts

```diff
@@ -80,11 +83,24 @@ export abstract class ComponentBase extends Disposable implements IComponent, On
 	public refreshDataProvider(item: any): void {
 	}
 
+	public updateStyles() {
+		this._CSSStyles = this.CSSStyles;
+		let styles = [];
+		for (var styleName in this._CSSStyles) {
+			styles.push(`${styleName}:${this._CSSStyles[styleName]}; `);
+		}
+		let element = <HTMLElement>this._el.nativeElement;
+		element.setAttribute('style', styles.join(''));
```

Comment:
Will this overwrite any additional styles? Is that the expected behavior and in what cases could it go wrong? E.g. what if they explicitly set a couple of styles that are per-component type?

Reply:
Thanks for catching this. I found a better solution by using builder

---
In reply to: [215820124](https://github.com/Microsoft/sqlopsstudio/pull/2454#discussion_r215820124) [](ancestors = 215820124)